django_features/system_message/viewsets.py

```diff
@@ -71,5 +73,7 @@ class SystemMessageViewSet(ModelViewSet):
 
     @action(methods=["patch"], detail=True)
     def dismiss(self, request: Request, pk: int, *args: Any, **kwargs: Any) -> Response:
+        if not self.request.user.is_authenticated:
+            Response(status=status.HTTP_204_NO_CONTENT)
         self.get_object().dismissed_users.add(self.request.user)
         return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
Aren't you mssing a return in the if statement? If you invert the if statement you only need one return.

```suggestion
        if self.request.user.is_authenticated:
            self.get_object().dismissed_users.add(self.request.user)
        return Response(status=status.HTTP_204_NO_CONTENT)
```